CMakeLists.txt

```diff
@@ -318,7 +318,15 @@ else()
     usFunctionCheckCompilerFlags(-Wno-mismatched-tags US_CXX_FLAGS)
   endif()
 
-  usFunctionCheckCompilerFlags("-fvisibility=hidden -fvisibility-inlines-hidden" _have_visibility)
+  # Issue #56: Workaround a clang compiler issue (https://llvm.org/bugs/show_bug.cgi?id=26749).
+  # The typeinfo symbols do not have reliable symbol visibility. Instead of exporting all
+  # symbols only export the typeinfo symbols by default. See
+  # https://developer.apple.com/library/mac/technotes/tn2185/_index.html#//apple_ref/doc/uid/DTS10004200-CH1-SUBSECTION2
+  if(US_COMPILER_CLANG OR US_COMPILER_APPLE_CLANG)
```

Comment:
The `US_COMPILER_*` variables denote only the compiler itself, with no reference to the platform they run on. As far as I know, the only officially supported compiler on MacOS since a few years is Apples Clang build which is different from upstream and also has a different versioning scheme. The LLVM website does not offer binaries for MacOS as far as I can see.

Since we still support CMake < 3.0, the variable `US_COMPILER_CLANG` will be set instead of `US_COMPILER_APPLE_CLANG` when using CMake 2.8.12 (the latter will be set to `1` when using a newer CMake version). This means we would actually need to use

`if(APPLE AND (US_COMPILER_CLANG OR US_COMPILER_APPLE_CLANG))`